php/libraries/SinglePointLogin.class.inc

```diff
@@ -316,6 +317,20 @@ class SinglePointLogin
             $this->_lastError = "Incorrect username or password.";
             return false;
         }
+        /* Check whether a user's account is locked due to too many bad login
+         * attempts before actually trying to authenticate their credentials.
+         * Doing so can help mititgate timing attacks against the server.
+         */
+        if ($this->accountLocked($_POST['username'])) {
+            $this->_lastError = 'This account is currently suspended due '
+                . 'to too many bad login attempts.';
+            $this->insertFailedDetail('Account locked', $setArray);
+            error_log(
```

Comment:
This isn't an error and doesn't belong in the error log

Reply:
I'll remove it. In another context I think we should discuss creating a log for security-related events. Without detailed logging for high-value transactions it'll be difficult to do any kind of incident response in the future.